pypeit/coadd2d.py

```diff
@@ -1428,9 +1510,9 @@ def get_maskdef_dict(self, slit_idx, ref_trace_stack):
                 self.stack_dict['slits_list'][0].maskdef_objpos is not None and \
                 self.stack_dict['maskdef_designtab_list'][0] is not None:
             # maskdef_designtab info for only this slit
-            this_idx = self.stack_dict['maskdef_designtab_list'][0]['SPAT_ID'] == self.stack_dict['slits_list'][0].spat_id[slit_idx]
+            this_idx = self.stack_dict['maskdef_designtab_list'][0]['SPAT_ID'] == self.spat_ids[slit_idx]
```

Comment:
Should the matching being done here also be within a tolerance set by spat_toler?